GVRf/Framework/framework/src/main/java/org/gearvrf/GVRFrustumPicker.java

```diff
@@ -0,0 +1,262 @@
+     * If the frustum is not attached to a scene object, it defaults to
+     * the view frustum of the main camera of the scene.
+     *
+     * @param frustum array of 6 floats as follows:
+     *                frustum[0] = left corner of frustum
+     *                frustum[1] = bottom corner of frustum
+     *                frustum[2] = front corner of frustum (near plane)
+     *                frustum[3] = right corner of frustum
+     *                frustum[4] = top corner of frustum
+     *                frustum[5 = back corner of frustum (far plane)
+     */
+    public void setFrustum(float[] frustum)
+    {
+        Matrix4f projMatrix = new Matrix4f();
+        projMatrix.setFrustum(frustum[0], frustum[3], frustum[1], frustum[4], frustum[2], frustum[5]);
+        setFrustum(projMatrix);
+    }
+
+    /**
+     * Set the view frustum to pick against from the field of view, aspect
+     * ratio and near, far clip planes. The viewpoint of the frustum
+     * is the center of the scene object the picker is attached to.
+     * The view direction is the forward direction of that scene object.
+     * The frustum will pick what a camera attached to the scene object
+     * with that view frustum would see. If the frustum is not attached
+     * to a scene object, it defaults to the view frustum of the main camera of the scene.
+     *
+     * @param fovy  vertical field of view in degrees
+     * @param aspect aspect ratio (width / height)
+
+     */
+    public void setFrustum(float fovy, float aspect, float znear, float zfar)
+    {
+        Matrix4f projMatrix = new Matrix4f();
+        projMatrix.perspective(fovy, (float) Math.toRadians(aspect), znear, zfar);
+        setFrustum(projMatrix);
+    }
+
+    /**
+     * Set the view frustum to pick against from the given projection  matrix.
+     *
+     * If the projection matrix is null, the picker will revert to picking
+     * objects that are visible from the viewpoint of the scene's current camera.
+     * If a matrix is given, the picker will pick objects that are visible
+     * from the viewpoint of it's owner the given projection matrix.
+     *
+     * @param projMatrix 4x4 projection matrix or null
+     * @see GVRScene.setPickVisible
+     */
+    public void setFrustum(Matrix4f projMatrix)
+    {
+        if (projMatrix != null)
+        {
+            if (mProjMatrix == null)
+            {
+                mProjMatrix = new float[16];
+            }
+            mProjMatrix = projMatrix.get(mProjMatrix, 0);
+            mScene.setPickVisible(false);
+        }
+        else
+        {
+            mScene.setPickVisible(true);
+        }
+        mProjection = projMatrix;
+        if (mCuller != null)
+        {
+            mCuller.set(projMatrix);
+        }
+        else
+        {
+            mCuller = new FrustumCuller(projMatrix);
+        }
+    }
+
+    public void onDrawFrame(float frameTime)
+    {
+        if (isEnabled())
```

Comment:
I think we should unregister the drawFrameListener if the component is disabled instead. Not something that needs immediate attention.